spring-pulsar/src/main/java/org/springframework/pulsar/core/reactive/ReactivePulsarSenderTemplate.java

```diff
@@ -0,0 +1,221 @@
+ *
+ * @param <T> the message payload type
+ * @author Christophe Bornet
+ */
+public class ReactivePulsarSenderTemplate<T> implements ReactivePulsarSenderOperations<T>, BeanNameAware {
+
+	private final LogAccessor logger = new LogAccessor(this.getClass());
+
+	private final ReactivePulsarSenderFactory<T> reactiveMessageSenderFactory;
+
+	private final ObservationRegistry observationRegistry;
+
+	private final PulsarTemplateObservationConvention observationConvention;
+
+	private String beanName;
+
+	private Schema<T> schema;
+
+	/**
+	 * Construct a template instance.
+	 * @param reactiveMessageSenderFactory the factory used to create the backing Pulsar
+	 * reactive sender.
+	 */
+	public ReactivePulsarSenderTemplate(ReactivePulsarSenderFactory<T> reactiveMessageSenderFactory) {
+		this(reactiveMessageSenderFactory, null, null);
+	}
+
+	/**
+	 * Construct a template instance with observation configuration.
+	 * @param reactiveMessageSenderFactory the factory used to create the backing Pulsar
+	 * reactive senders
+	 * @param observationRegistry the registry to record observations with or {@code null}
+	 * to not record observations
+	 * @param observationConvention the optional custom observation convention to use when
+	 * recording observations
+	 */
+	public ReactivePulsarSenderTemplate(ReactivePulsarSenderFactory<T> reactiveMessageSenderFactory,
+			@Nullable ObservationRegistry observationRegistry,
+			@Nullable PulsarTemplateObservationConvention observationConvention) {
+		this.reactiveMessageSenderFactory = reactiveMessageSenderFactory;
+		this.observationRegistry = observationRegistry;
+		this.observationConvention = observationConvention;
+	}
+
+	@Override
+	public Mono<MessageId> send(T message) {
+		return doSend(null, message, null, null, null);
+	}
+
+	@Override
+	public Mono<MessageId> send(String topic, T message) {
+		return doSend(topic, message, null, null, null);
+	}
+
+	@Override
+	public ReactiveSendMessageBuilderImpl<T> newMessage(T message) {
+		return new ReactiveSendMessageBuilderImpl<>(this, message);
+	}
+
+	@Override
+	public void setBeanName(String beanName) {
+		this.beanName = beanName;
+	}
+
+	/**
+	 * Set the schema to use on this template.
+	 * @param schema provides the {@link Schema} used on this template
+	 */
+	public void setSchema(Schema<T> schema) {
+		this.schema = schema;
+	}
+
+	private Mono<MessageId> doSend(String topic, T message,
+			MessageSpecBuilderCustomizer<T> messageSpecBuilderCustomizer, MessageRouter messageRouter,
+			ReactiveMessageSenderBuilderCustomizer<T> customizer) {
+		final String topicName = ReactiveMessageSenderUtils.resolveTopicName(topic, this.reactiveMessageSenderFactory);
+		this.logger.trace(() -> String.format("Sending reative msg to '%s' topic", topicName));
+
+		PulsarMessageSenderContext senderContext = PulsarMessageSenderContext.newContext(topicName, this.beanName);
+		Observation observation = newObservation(senderContext);
+		try {
+			observation.start();
```

Comment:
If we did leave this in (see below comment) I wonder if the start call should go in the `doSubscribe` callback rather than outside the reactive chain. I would want to see what the Spring Framework is doing for WebClient observation instrumentation.

Reply:
Observations has been removed from now.
Will wait till it's hardened on the imperative side.